src/ExUnitRunner.ts

```diff
@@ -0,0 +1,101 @@
+import path = require('path');
+import * as fs from 'fs';
+import { TestSuiteInfo } from 'vscode-test-adapter-api';
+import { MixParser } from './MixParser';
+import { MixRunner } from './MixRunner';
+import { TestTree } from './TestTree';
+import { parseMixOutput } from './utils/tests_parsers';
+
+export interface TestResult {
+  state: 'failed' | 'passed';
+  nodeId: 'string';
+  error?: 'string';
+}
+
+/*
+ This class is responsible for loading and running tests.
+ Under the hood it uses the TestTree and the MixRunner to do so.
+*/
+export class ExUnitRunner {
+  private readonly workspaceName: string;
+  private readonly workspacePath: string;
+  private mixRunner: MixRunner;
+  private mixParser: MixParser;
+  private testTree: TestTree;
+
+  constructor(workspaceName: string, workspacePath: string) {
+    this.mixRunner = new MixRunner();
+    this.mixParser = new MixParser();
+    this.testTree = new TestTree(workspaceName);
+    this.workspaceName = workspaceName;
```

Comment:
this seems unnecessary? It's not used anywhere in this file.

Reply:
Removed!